main.go

```diff
@@ -15,6 +15,108 @@
 
 package main
 
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/memfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/google/go-github/v60/github"
+	"github.com/stacklok/frizbee-action/pkg/action"
+	"github.com/stacklok/frizbee/pkg/replacer"
+	"github.com/stacklok/frizbee/pkg/utils/config"
+
+	"golang.org/x/oauth2"
+	"log"
+	"os"
+	"strings"
+)
+
 func main() {
-	// This is a placeholder for the main function
+	ctx := context.Background()
+	// Initialize the frizbee action
+	frizbeeAction, err := initAction(ctx)
+	if err != nil {
+		log.Fatalf("Error initializing action: %v", err)
+	}
+
+	// Run the frizbee action
+	err = frizbeeAction.Run(ctx)
+	if err != nil {
+		if errors.Is(err, action.ErrUnpinnedFound) {
+			log.Printf("Unpinned actions or container images found. Check the Frizbee Action logs for more information.")
+			os.Exit(1)
+		}
+		log.Fatalf("Error running action: %v", err)
+	}
+}
+
+// initAction initializes the frizbee action - reads the environment variables, creates the GitHub client, etc.
+func initAction(ctx context.Context) (*action.FrizbeeAction, error) {
+	// Get the GitHub token from the environment
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		return nil, fmt.Errorf("GITHUB_TOKEN environment variable is not set")
+	}
+
+	// Create a new GitHub client
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(ctx, ts)
+
+	// Get the GITHUB_REPOSITORY_OWNER
+	repoOwner := os.Getenv("GITHUB_REPOSITORY_OWNER")
+	if repoOwner == "" {
+		return nil, fmt.Errorf("GITHUB_REPOSITORY_OWNER environment variable is not set")
+	}
+
+	// Split the GITHUB_REPOSITORY environment variable to get repo name
+	repoFullName := os.Getenv("GITHUB_REPOSITORY")
+	if repoFullName == "" {
+		return nil, fmt.Errorf("GITHUB_REPOSITORY environment variable is not set")
+	}
+
+	// Clone the repository
+	fs, repo, err := cloneRepository(fmt.Sprintf("https://github.com/%s", repoFullName), repoOwner, token)
```

Comment:
not sure if there's any use-case, but we might want to use the GH API to get the clone URL rather than constructing it ourselves.